File: solutions/typescript/acronym/1/acronym.ts
export function parse(phrase: string): string {
  return phrase
    .replace(/([a-z])([A-Z])/g, '$1 $2') 
    .replace(/-/g, ' ')
    .replace(/[^a-zA-Z\s]/g, '')      
    .split(/\s+/)
    .filter(word => word.length > 0)  
    .map(word => word[0].toUpperCase())
    .join('');
}

File: solutions/kotlin/acronym/1/src/main/kotlin/Acronym.kt
object Acronym {
    fun generate(phrase: String): String {
        return phrase
            .replace("-", " ")
.filter { it.isLetter() || it.isWhitespace() }
           
            .split(" ")
            .filter { it.isNotEmpty() }
  
            .map { it.first().uppercase() }

            .joinToString("")
    }
}
